security/advancedtls/testdata/server_cert_3.pem

```diff
@@ -0,0 +1,22 @@
+-----BEGIN CERTIFICATE-----
```

Comment:
Several comments for the generated certificate:
1. we might want to set longer validity for the test certificate, maybe 20 years or so. Otherwise it could cause some unexpected problems when it's expired after a year.
2. Ideally we want to see SNI works for both Common Name and the DNS field under SAN. Since we already had certificates with some common names, can we set some DNS names under the SAN field of this certificate? To learn the difference of these two, you can refer to: [this link](https://support.dnsimple.com/articles/what-is-common-name/#:~:text=on%20Google%20Safari-,Common%20Name%20vs%20Subject%20Alternative%20Name,wildcard%20or%20non%2Dwildcard%20name.&text=The%20SAN%20allows%20issuance%20of,Authority%20and%20the%20specific%20product.)

Reply:
Sure, I can generate another certificate with DNS names.

Reply:
Update: removed the old certificate pair `server_cert_3.pem` and `server_key_3.pem` which was a certificate with common name setting only. Generated a new certificate pair with the same name `server_cert_3.pem` and `server_key_3.pem`, but had both common name and DNS names set.
In addition, I also added a new test case to test if SNI works for DNS field of the newly added certificate pair.

Reply:
Might be worth adding the human readable version of the cert (as generated by the openssl command) either to the same file (if it supports comments), or add it as a separate file. Something like `server_cert_3.txt`.